kit/Kit.cpp

```diff
@@ -1083,16 +1090,22 @@ void Document::trimAfterInactivity()
 
     LOG_TRC("Should we trim our caches ?");
     double minInactivityMs = std::numeric_limits<double>::max();
+    bool allClientsDeclaredInactivity = true;
     for (const auto& it : _sessions)
     {
         minInactivityMs = std::min(it.second->getInactivityMS(), minInactivityMs);
+        if (it.second->isActive())
+            allClientsDeclaredInactivity = false;
     }
 
     if (minInactivityMs >= 9999)
     {
-        LOG_DBG("Trimming Core caches");
+        LOG_DBG("Trimming Core caches: " << (allClientsDeclaredInactivity ? "deep" : "shallow"));
         SigUtil::addActivity("trimAfterInactivity");
-        _loKit->trimMemory(1024);
+        if (allClientsDeclaredInactivity)
```

Comment:
Problem is - stats show that our ~90% use-case is just one view; so this seems to mean we do a deep trim ~always instead of a shallow one (?)

I'd prefer to wait some timeout after we get back to not running a trim due to a background save and check again for idling if possible.

Reply:
We currently do a deep trim in trimIfInactive (triggered from an explicit "userinactive" from online that sets a session's isActive to false) when all the sessions have that isActive bit off.

The thought here was to make the other regularly run trimAfterInactivity also deep if all sessions are in that "userinactive" mode. So things would go from:

trimAfterInactivity(s): shallow trim
<userinactive>
trimIfInactive: deep trim (maybe skipped)
trimAfterInactivity(s): shallow trim
<useractive>
trimAfterInactivity(s): shallow trim

to

trimAfterInactivity(s): shallow trim
<userinactive>
trimIfInactive: deep trim (maybe skipped)
trimAfterInactivity(s): deep trim
<useractive>
trimAfterInactivity(s): shallow trim